src/main/java/uk/gov/hmcts/cmc/claimstore/services/SettlementAgreementService.java

```diff
@@ -31,31 +36,35 @@
     private final CaseRepository caseRepository;
     private final EventProducer eventProducer;
     private final AppInsights appInsights;
+    private final CCDEventProducer ccdEventProducer;
 
     @Autowired
     public SettlementAgreementService(
         ClaimService claimService,
         CaseRepository caseRepository,
         EventProducer eventProducer,
-        AppInsights appInsights
+        AppInsights appInsights,
+        CCDEventProducer ccdEventProducer
     ) {
         this.claimService = claimService;
         this.caseRepository = caseRepository;
         this.eventProducer = eventProducer;
         this.appInsights = appInsights;
+        this.ccdEventProducer = ccdEventProducer;
     }
 
     public Claim reject(Claim claim, String authorisation) {
 
         Settlement settlement = assertSettlementCanBeResponded(claim);
         settlement.reject(MadeBy.DEFENDANT);
 
-        String userAction = format("SETTLEMENT_AGREEMENT_REJECTED_BY_%s", MadeBy.DEFENDANT.name());
+        String userAction = AGREEMENT_REJECTED_BY_DEFENDANT.getValue();
```

Comment:
Please do not to use CCD event names here in this context. Please revert these changes back.

Reply:
Not following your comment. I will call you

Reply:
sabah, these event names are just for ccd only, on db call it is ignored

Reply:
Please ignore above, but try to use enum rather than `getValue()` where ever possible.

Reply:
@sabahirfan the value matches the events in CCD, currently testing it.

Reply:
We are passing string value here and then in `CCDCaseRepository` converting it back to enum by calling `CaseEvent.valueOf(userAction)`. Would be clean to pass enum and avoid unnecessary conversion. 

Here we are converting it back to enum.
https://github.com/hmcts/cmc-claim-store/blob/585e6a1491c36fef5b66e1a0d90bdab598cb3a3b/src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseRepository.java#L135

Reply:
@sabahirfan I saw this already. I am already refactoring it to send it as Enum.